sklearn/manifold/_isomap.py

```diff
@@ -145,6 +154,7 @@ def __init__(self, *, n_neighbors=5, n_components=2, eigen_solver='auto',
 
     def _fit_transform(self, X):
```

Comment:
I think we should raise either a `ValueError` or a `UserWarning` with an explicit error message when both `n_neighbors` and `radius` are not `None`.

- if `ValueError`, explain that we should either path `n_neighbors` or `radius` and the other should be set to `None` (and recall the observed values in the message).
- if `UserWarning`, explain that `radius` is ignored and that `n_neighbors` should be set to `None` explicitly when `radius` is passed.

I think I would favor option the `ValueError` option. And opinion @TomDLT?

In any case we need a test to check the error message.

Reply:
I have no preference here, but I agree we have to add something.

Reply:
To me, a `UserWarning`, as currently implemented, suffices.